index.js

```diff
@@ -1 +1,207 @@
-console.log('hello world');
+import methodOverride from 'method-override';
+import express from 'express';
+import {
+  read, add, deleteContent, edit,
+} from './jsonFileStorage.js';
+
+const app = express();
+const PORT = 3000;
+const FILENAME = 'data.json';
+
+// Set view engine to EJS
+app.set('view engine', 'ejs');
+// Middleware to allow static images/css files to be served
+app.use(express.static('public'));
+// Provides encoding to url queries etc.
+app.use(express.urlencoded({ extended: false }));
+// Middleware that allows POST methods to be overriden to perform PUT and DELETE requests
+app.use(methodOverride('_method'));
+
+// Helper Function that converts request.body to an Iterable Obj
+const convertToIterableObj = (nonIterableObj) => {
+  const newObj = {};
+  // Convert request.body into an iterable object
+  Object.entries(nonIterableObj).forEach(([key, value]) => {
+    newObj[key] = value;
+  });
+  return newObj;
+};
+
+// Helper Function that checks the inputs for validation
+const performValidationOnRequestBody = (requestBodyObj, callback) => {
+  const inputValidationFeedbackObj = {};
+  let isFormValid = true;
+  const {
+    _, description, date_time, city, state, shape, duration, summary, __,
+  } = requestBodyObj;
+
+  if (description.length < 5) {
+    inputValidationFeedbackObj.description = 'Description is too short!';
+    isFormValid = false;
+  } else {
+    inputValidationFeedbackObj.description = 'ok';
+  }
+
+  if (shape.toLowerCase() !== 'sphere' && shape.toLowerCase() !== 'circle') {
+    inputValidationFeedbackObj.shape = 'Shape is not valid.';
+    isFormValid = false;
+  } else {
+    inputValidationFeedbackObj.shape = 'ok';
+  }
+  callback(inputValidationFeedbackObj, isFormValid);
+};
+
+// Route: Homepage - Shows all sightings, sortable by ID
+app.get('/', (request, response) => {
+  const orderOfSort = request.query.sortOrder;
+  read(FILENAME, (data, error) => {
+    if (error) {
+      response.sendStatus(500, error);
+      return;
+    }
+    data.sightings.sort((a, b) => {
+      if (orderOfSort === 'ascending') {
+        return a.id - b.id;
+      } if (orderOfSort === 'descending') {
+        return b.id - a.id;
+      }
+      return a.id - b.id;
+    });
+    response.render('allSightings', data);
+  });
+});
+
+// Route: Create form to input sighting
+app.get('/newSighting', (request, response) => {
+  read(FILENAME, (data, error) => {
+    if (error) {
+      response.sendStatus(500, error);
+      return;
+    }
+    response.render('submitNewSightingForm', data);
+  });
+});
+
+// Create a POST method / request to submit new sighting
+app.post('/newSighting', (request, response) => {
+  const requestBodyObj = convertToIterableObj(request.body);
+  // locally scoped boolean to track if current form is valid
+  let isThisFormValid = true;
+
+  // Perform input validation first
+  performValidationOnRequestBody(requestBodyObj, (feedbackMsg, areFieldsValid) => {
```

Comment:
Instead of using a callback here, perhaps a clearer architecture would be to have this function return a boolean, and call the function something like `validateFormFields`, where it returns `true` if fields are valid and `false` otherwise. 

Then we can do something like:
```
const areFormFieldsValid = validateFormFields(requestBodyObj);

if (!areFormFieldsValid) {
  request.body.feedback = feedbackMsg;
  response.render('submitNewSightingForm', request.body);
  return;
}